src/annotatedyaml/loader.py

```diff
@@ -256,13 +252,13 @@ def _add_reference(
     """Add file reference information to an object."""
     if isinstance(obj, list):
         obj = NodeListClass(obj)
-        _add_reference_to_node_list_class(obj, loader, node)
     elif isinstance(obj, str):
         obj = NodeStrClass(obj)
-        _add_reference_to_node_str_class(obj, loader, node)
     elif isinstance(obj, dict):
         obj = NodeDictClass(obj)
-        _add_reference_to_node_dict_class(obj, loader, node)
+    else:
+        return obj  # type: ignore[unreachable]
```

Comment:
The typing on this is too narrow. It can actually be `int` as well but I didn't want to change that here since its not the substance of this PR other than the references being added.